modules/oke/nodepools.tf

```diff
@@ -21,7 +21,8 @@ resource "oci_containerengine_node_pool" "nodepools" {
     # iterating over ADs
     dynamic "placement_configs" {
       iterator = ad_iterator
-      for_each = local.ad_names
+      for_each = [for n in lookup(each.value, "placement_ads", local.ad_numbers):
+                  ad_number_to_name[n]]
```

Comment:
This should be referenced as local.ad_number_to_name) once you've figured out how to resolve the above problem